support-services/src/main/scala/com/gu/support/promotions/PromotionCollection.scala

```diff
@@ -19,11 +19,12 @@ class CachedDynamoPromotionCollection(config: PromotionsTablesConfig)
     extends DynamoPromotionCollection(config)
     with PromotionCollection {
 
-  override def all: Iterator[Promotion] = PromotionCache.get.getOrElse(fetchAndCache).toIterator
+  val cache = new PromotionCache
```

Comment:
have you used printlns or something just to check that only two PromotionCollection are created during the app lifecycle?  I think it is ok, but it would be annoying if it was one per request or something!

Reply:
haveing searched about a bit I'm a bit concerned why we have 3 places creating promotion service - see the "non-test" versions in this search
https://github.com/search?q=repo%3Aguardian%2Fsupport-frontend+%22new+PromotionService%22&type=code

Reply:
> have you used printlns or something

<img width="454" alt="Screenshot 2024-02-06 at 16 31 24" src="https://github.com/guardian/support-frontend/assets/31692/a823bfb8-ac68-477b-b223-2d7969b04ac7">

For piece of mind on `support-frontend` - this is all I got from multiple refreshes on the app once started.

Reply:
I think the intention though is that every instance of PromotionsService has it's own cache.

We could probably make this more scalable, buy having a keyed cache e.g.
```scala
val cache: Map[Env, Promotions] = ???
def get(env: Env) = cache.get(env)
```

But I'm not sure our use-case warrants this complexity.

Reply:
sounds good thanks for the updates (sorry I was probably editing my comment while you were already reading it).  If we wanted to be all belt and braces about it in case of regressions, we could keep a counter and alarm if it goes above 10, but I don't think it's worth it.

Reply:
Yea, or you could have the cache generated at the start up level and injected into the services to avoid it all-together... Let's give it a whirl and see if it becomes a problem.

Reply:
I'm going to leave it as is an if we fancy, we can try flesh out a better abstraction for this.